### PR TITLE
Fix an issue where wrapper exceptions were not being applied correctly

### PR DESCRIPTION
**Pull request by achocron**
After all of the refactoring I missed a serious issue where the wrapper exceptions were not being applied to the error logs properly. This PR fixes that (once and for all)!

### CODE REVIEW

File: MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm [hunk 1]
```diff
@@ -636,24 +636,20 @@ - (void)processCrashReports {
...
 
   // First save all found crash reports for use in correlation step.
   NSMutableDictionary *foundCrashReports = [[NSMutableDictionary alloc] init];
-  NSMutableDictionary *foundErrorLogs = [[NSMutableDictionary alloc] init];
   NSMutableDictionary *foundErrorReports = [[NSMutableDictionary alloc] init];
   for (NSURL *fileURL in self.crashFiles) {
     NSData *crashFileData = [NSData dataWithContentsOfURL:fileURL];
     if ([crashFileData length] > 0) {
       MSPLCrashReport *report = [[MSPLCrashReport alloc] initWithData:crashFileData error:&error];
-      MSAppleErrorLog *log = [MSErrorLogFormatter errorLogFromCrashReport:report];
```
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm]: Is it still okay for all native crash logs??

Comment (achocron) [inline MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm]: Yes we just create the logs later

Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm]: Now understand what you are doing here. 😄

File: MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm [hunk 2]
```diff
@@ -779,6 +775,7 @@ - (void)handleLatestCrashReport {
         NSURL *cacheURL = [self.crashesDir URLByAppendingPathComponent:cacheFilename];
         [crashData writeToURL:cacheURL atomically:YES];
         self.lastSessionCrashReport = [MSErrorLogFormatter errorReportFromCrashReport:report];
+        [MSWrapperExceptionManager correlateLastSavedWrapperExceptionToReport:@[self.lastSessionCrashReport]];
```
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm]: This line brought up some design concerns that are not related to your PR. We can talk offline.
